### PR TITLE
feat: remove `enabled` from table `connectors`

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

remove `enabled` from table `connectors`

## Testing

Passed UTs and ITs.

### CODE REVIEW

File: packages/core/src/lib/passcode.ts [hunk 1]
```diff
@@ -65,8 +65,10 @@ export const sendPasscode = async (passcode: Passcode) => {
   );
 
   const { dbEntry, metadata, sendMessage } = connector;
+  console.log('dbEntry:', dbEntry, dbEntry.config);
```
Comment (wangsijie) [inline packages/core/src/lib/passcode.ts:68]: remove this line

File: packages/core/src/lib/passcode.ts [hunk 2]
```diff
@@ -65,8 +65,10 @@ export const sendPasscode = async (passcode: Passcode) => {
...
 
   const messageTypeResult = messageTypesGuard.safeParse(passcode.type);
+  console.log('messageTypeResult:', messageTypeResult);
```
Comment (wangsijie) [inline packages/core/src/lib/passcode.ts:71]: remove this line

File: packages/core/src/lib/sign-in-experience/index.test.ts [hunk 3]
```diff
@@ -4,9 +4,9 @@ import type { CreateSignInExperience, SignInExperience } from '@logto/schemas';
 import { BrandingStyle } from '@logto/schemas';
 
 import {
-  disabledSocialTarget01,
-  disabledSocialTarget02,
-  enabledSocialTarget01,
+  socialTarget01,
```
Comment (wangsijie) [inline packages/core/src/lib/sign-in-experience/index.test.ts:7]: The disabled target can be just removed?

File: packages/core/src/routes/connector.test.ts [hunk 4]
```diff
@@ -58,13 +59,13 @@ describe('connector route', () => {
       jest.clearAllMocks();
     });
 
-    it('throws if more than one email connector is enabled', async () => {
+    it('throws if more than one email connector is added', async () => {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.test.ts:62]: nit: exists or available

File: packages/core/src/routes/connector.test.ts [hunk 5]
```diff
@@ -188,6 +189,39 @@ describe('connector route', () => {
       });
       expect(response).toHaveProperty('statusCode', 422);
     });
+
+    it('should post a new record when add more than 1 instance with connector factory', async () => {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.test.ts:193]: This belongs to another PR?

Comment (wangsijie) [reply packages/core/src/routes/connector.test.ts:193]: Alright, this "enabled" check is now on "post" api, but this description is confusing

File: packages/core/src/routes/connector.ts [hunk 6]
```diff
@@ -122,66 +117,28 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
-
-      // Only allow one enabled connector for SMS and Email.
-      // disable other connectors before enable this one.
-      if (enabled && (type === ConnectorType.Sms || type === ConnectorType.Email)) {
-        const connectors = await getLogtoConnectors();
+      if (
+        connectorFactory.type === ConnectorType.Sms ||
+        connectorFactory.type === ConnectorType.Email
+      ) {
+        const logtoConnectors = await getLogtoConnectors();
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:130]: can we keep the same logic for "Only allow one enabled connector for SMS and Email."? This PR is aim to remove "enabled", try to keep other things unchanged.

File: packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ import type { AlterationScript } from '../lib/types/alteration.js';
 const alteration: AlterationScript = {
   up: async (pool) => {
     await pool.query(sql`
+      ALTER TABLE connectors DROP COLUMN enabled;
```
Comment (wangsijie) [inline packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts:8]: If we want to remove this column, better remove all "disabled" rows?

File: packages/core/src/routes/connector.ts [hunk 8]
```diff
@@ -131,61 +126,6 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
-      }
-
-      const connector = await updateConnector({
-        set: { enabled },
-        where: { id },
-        jsonbMode: 'merge',
-      });
-
-      // Delete the social connector in the sign-in experience if it is disabled.
-      if (!enabled && type === ConnectorType.Social) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:179]: need to implement this on "delete" API before merging this PR?

File: packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts [hunk 9]
```diff
@@ -5,6 +5,8 @@ import type { AlterationScript } from '../lib/types/alteration.js';
 const alteration: AlterationScript = {
   up: async (pool) => {
     await pool.query(sql`
+      DELETE FROM connectors WHERE enabled = false;
```
Comment (wangsijie) [inline packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts:8]: This alteration is merged into master, is not forbidden to modify, please create a new alteration script.

**Review by wangsijie — CHANGES_REQUESTED** [review]

File: packages/core/src/routes/connector.ts (hunk 8 above)
Comment (darcyYe) [reply packages/core/src/routes/connector.ts:179]: Move this logic to [this PR](https://github.com/logto-io/logto/pull/2524).

File: packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts (hunk 9 above)
Comment (darcyYe) [reply packages/schemas/alterations/next-1668666590-support-multiple-connector-instances.ts]: Move them to a new script.

**Comment by darcyYe** [comment]
Base branch need to pull origin master.

**Comment by charIeszhao** [comment]
Please rebase the PR to include only your commits

**Comment by darcyYe** [comment]
Rebased base branch, but it still shows others' commits on this page.
image
image

**Review by wangsijie — APPROVED** [review]
